rs/tests/execution/general_execution_tests/nns_shielding.rs

```diff
@@ -81,7 +81,7 @@ pub fn mint_cycles_supported_only_on_cycles_minting_canister(env: TestEnv) {
                         "Error from Canister {}: Canister violated contract: ic0.mint_cycles cannot be executed on non Cycles Minting Canister: {} != {}.\nThis is likely an error with the compiler/CDK toolchain being used to build the canister. Please report the error to IC devs on the forum: https://forum.dfinity.org and include which language/CDK was used to create the canister.",
                         nns_canister_id, nns_canister_id,
                         CYCLES_MINTING_CANISTER_ID),
-                    error_code: None})
+                    error_code: Some("IC0504".to_string())})
```

Comment:
This error code started being returned. Is this expected?

Reply:
Yes, same for the rest. These are the right error codes for the test case, my guess is that the previous version of the agent was not populating the `error_code` field (only the `reject_code` was being populated).